### ISSUE TITLE
Unity scene parsing example?

### ISSUE THREAD

**Issue by soraryu** [aaubry/YamlDotNet#307]
Since you (@aaubry) are often writing about "Unity support" would be great to see a working example of parsing a unity scene file YAML. I don't get it to work.

Here's what I tried, and I don't know what's wrong:

```        var input = new StringReader(Document);
        var output = new StringBuilder();
        var deserializer = new DeserializerBuilder()
            .IgnoreUnmatchedProperties()
            .Build();
        var parser = new Parser(input);
        // Consume the stream start event "manually"
        parser.Expect<StreamStart>();

        while (parser.Accept<DocumentStart>())
        {
            Debug.Log(parser.Current.ToString());

            try
            {
                // Deserialize the document
                var doc = deserializer.Deserialize<FilePart>(parser);

                output.AppendLine("## Document : ");

                if (doc.OcclusionCullingSettings != null)
                {
                    var go = doc.OcclusionCullingSettings;
                    output.AppendLine(go.serializedVersion + ", ");
                }
                
                if (doc.RenderSettings != null)
                {
                    var go = doc.RenderSettings;
                    output.AppendLine(go.serializedVersion + ", ");
                }

                if (doc.GameObject != null)
                {
                    var go = doc.GameObject;
                    output.AppendLine(go.m_Name + ", " + go.serializedVersion);
                }
            }
            catch(System.Exception e)
            {
                Debug.Log(e.ToString());
            }
        }

        Debug.Log(output.ToString());```

With the following base classes:
```[System.Serializable]
    public class FilePart
    {
        public BaseDescription OcclusionCullingSettings;
        public BaseDescription RenderSettings;
        public GameObjectDescription GameObject;
        public BaseDescription MonoBehaviour;
    }
    [System.Serializable]
    public class BaseDescription
    {
        public int m_ObjectHideFlags;
        public int serializedVersion;
        public BaseDescription()
        {        }
    }
    [System.Serializable]
    public class GameObjectDescription : BaseDescription
    {
        public string m_Name = "";
        public GameObjectDescription()
        {        }
    }```

**Comment by aaubry** [comment]
I have never used Unity, and I don't know much about scene files. The support that I mention is that the library is usable in Unity, not that you can parse scene files with it. The Unity files that [I have seen are not even valid YAML](https://github.com/aaubry/YamlDotNet/issues/140).

But if you provide more information, I may be able to help you. What behavior are you observing? Do you get an exception? Can you provide a full example, including the file that you are trying to parse?

**Comment by soraryu** [comment]
I guess I got a lot closer by adding that UnityNodeTypeResolver in – got rid of some exceptions, stuck with new ones and with not getting any data. But now I have the feeling I'm just lacking an understanding of how to use YamlDotNet.

I attached a zip file with two unity scenes (that are valid YAML as far as I can tell) and the relevant parsing code. I tried to strip everything Unity-related away, so you should be able to get these running in any environment (might have to replace the log call).

[ParseYamlScene.zip](https://github.com/aaubry/YamlDotNet/files/1836559/ParseYamlScene.zip)

**Comment by hanbim520** [comment]
Hi,unity yaml is not the standard yaml format (I think so), I achieved a resolution, but it is not perfect, not much time. Tested on 5.6.4, 2017 should not be a problem. You are interested in it. I hope everyone can improve this piece together.
https://github.com/hanbim520/UnityYaml.git

**Comment by aaubry** [comment]
I will close this issue as there has been no feedback. Feel free to reopen it if necessary.

**Comment by soraryu** [comment]
@aaubry actually it was your turn to reply :) I provided a full example as you requested and you wanted to take a look at. Did you?

**Comment by aaubry** [comment]
Humm, sorry, I was closing a bunch of old issues and this one was included by mistake.

I looked at the code that you provided but had trouble running it because it uses types such as MonoBehaviour, that I don't have available. I tried to remove that and run what I could but did not see any exception.
I don't even know what exactly is the problem that you are facing because you did not explain it.

If you want help with this, please provide either a **unit test** or a **console application** that reproduces your problem. Otherwise I'm left to guess what you need.

**Comment by HarmlessEvil** [comment]
@aaubry hello! I am also trying to parse some Unity scene files using this library.

I modified @soraryu's example (I used it as starting point) and made a representative console application.

The goal is to see in the console actual values of `serializedVersion` field, which is `2, 9, 11, …` and so on, according to the `Test01.unity`

In my case I see `0` for every deserialized node.

Here is the code for representation of issue: [issue.zip](https://github.com/aaubry/YamlDotNet/files/4402477/issue.zip). There are one scene and custom type resolver. I don't clearly understand how this should be implemented correctly yet.
